web_push/src/lib.rs

```diff
@@ -168,6 +172,11 @@ impl AddHeaders for () {
     }
 }
 
+// Just to get the `impl<A: AddHeaders> WebPushBuilder<A>` to build without the vapid feature
+#[doc(hidden)]
+#[cfg(not(feature = "vapid"))]
+pub trait AddHeaders: Sized {}
```

Comment:
This could probably be avoided with a bit more refactoring, but I went for the minimal implementation here.